packages/controllers/src/utils/SIWXUtil.ts

```diff
@@ -125,11 +125,9 @@ export const SIWXUtil = {
         })
       }
 
-      if (properties.isSmartAccount) {
-        SnackController.showError('This application might not support Smart Accounts')
-      } else {
-        SnackController.showError('Signature declined')
-      }
+      const message = error instanceof Error ? error.message : 'Signature declined'
+
+      SnackController.showError(message)
```

Comment:
do we want to show the full error? what happens when rejecting a request? 
have you tried shutting internet off to dforce viem/upstream service fail?